lib/active_record/connection_adapters/trilogy_adapter.rb

```diff
@@ -153,13 +153,87 @@ def discard!
         self.connection = nil
       end
 
+      # ActiveRecord 7.0 support
+      if ActiveRecord.version < ::Gem::Version.new('7.1.a')
+        def initialize(connection, logger, connection_options, config)
+          super
+          if @connection
+            @verified = true
+            @raw_connection = @connection
+          end
+          # Ensure that we're treating prepared_statements in the same way that Rails 7.1 does
+          @prepared_statements = self.class.type_cast_config_to_boolean(
+            @config.fetch(:prepared_statements) { default_prepared_statements }
+          )
+        end
+
+        def connect!
+          verify!
+          self
+        end
+
+        def reconnect!
+          @lock.synchronize do
+            disconnect!
+            connect
+          rescue StandardError => original_exception
+            @verified = false
+            raise translate_exception_class(original_exception, nil, nil)
+          end
+        end
+
+        def exec_rollback_db_transaction
+          # 16384 tests the bit flag for SERVER_SESSION_STATE_CHANGED, which gets set when the
+          # last statement executed has caused a change in the server's state.
+          if active? || (@raw_connection.server_status & 16_384).positive?
+            super
+          else
+            @verified = false
+          end
+        end
+
+        def with_raw_connection(uses_transaction: true, **_kwargs)
+          @lock.synchronize do
+            @raw_connection = @connection || nil unless instance_variable_defined?(:@raw_connection)
+            unless @verified
+              verify!
+              @verified = true
+            end
+            materialize_transactions if uses_transaction
+            begin
+              yield @raw_connection
+            rescue StandardError => e
+              @verified = false unless e.is_a?(Deadlocked) || e.is_a?(LockWaitTimeout) ||
+                                       ( # Timed out while in a transaction?
+                                         @raw_connection &&
+                                         (@raw_connection.server_status & 1).positive? &&
+                                         (e.is_a?(Trilogy::ClientError) || e.is_a?(Errno::ETIMEDOUT))
+                                       )
+              raise
+            end
+          end
+        end
+
+        def execute(sql, name = nil, **kwargs)
```

Comment:
Q: any reason why you are overriding this method ?

Reply:
This in conjunction with the way `reset!` works allows these three examples to pass:

#reset_closes_connection_with_existing_connection
#reset_answers_new_connection_with_existing_connection
#reset_answers_new_connection_without_existing_connection